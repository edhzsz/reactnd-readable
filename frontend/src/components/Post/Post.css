.post {
  width: 100%;
}

.post-title {
  font-size: 24px;
  font-weight: 300;
  padding: 20px 24px;
  margin: 0;
  color: #000;
}

.post-contents {
  min-height: 100px;
  background-color: #F5F5F6;
  color: #000;
  padding: 24px;
  overflow: hidden;
}

.post-main {
  float: left;
}

.post-body{
  padding-bottom: 5px;
  border-bottom: 1px solid grey;
}
