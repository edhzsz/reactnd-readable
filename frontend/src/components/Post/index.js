import React from 'react';
import PropTypes from 'prop-types';
import './Post.css'
import Votes from '../Votes';
import CommentsList from '../CommentsList'

const Post = ({post}) => {
  return (
    <div className="post">
      <div className="post-title base">
        <span>{post.title}</span>
      </div>
      <div className="post-contents">
        <div className="post-votes">
          <Votes votes={post.voteScore} />
        </div>
        <div className="post-main">
          <div className="post-body">
            {post.body}
          </div>
          <CommentsList comments={post.comments} />
        </div>
      </div>
    </div>
  );
}

Post.propTypes = {
  post: PropTypes.shape({
    id: PropTypes.string.isRequired,
    timestamp: PropTypes.number,
    title: PropTypes.string.isRequired,
    body: PropTypes.string,
    author: PropTypes.string,
    category: PropTypes.string,
    voteScore: PropTypes.number,
    comments: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.string.isRequired,
      author: PropTypes.string.isRequired,
      body: PropTypes.string.isRequired,
      timestamp: PropTypes.number,
      voteScore: PropTypes.number,
    }))
  }).isRequired
}

export default Post;