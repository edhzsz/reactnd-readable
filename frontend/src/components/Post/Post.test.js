import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Post from '.';

const testPost = {
  id: '8xf0y6ziyjabvozdd253nd',
  timestamp: 1467166872634,
  title: 'Udacity is the best place to learn React',
  body: 'Everyone says so after all.',
  author: 'thingtwo',
  category: 'react',
  voteScore: 6,
  deleted: false,
  comments: [
    {
      id: '894tuq4ut84ut8v4t8wun89g',
      parentId: "8xf0y6ziyjabvozdd253nd",
      timestamp: 1468166872634,
      body: 'Hi there! I am a COMMENT.',
      author: 'thingtwo',
      voteScore: 6,
      deleted: false,
      parentDeleted: false
    },
    {
      id: '8tu4bsun805n8un48ve89',
      parentId: "8xf0y6ziyjabvozdd253nd",
      timestamp: 1469479767190,
      body: 'Comments. Are. Cool.',
      author: 'thingone',
      voteScore: -5,
      deleted: false,
      parentDeleted: false
    }
  ]
};

const wrapper = shallow(<Post post={testPost}/>);

describe('Component: Post', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Post post={testPost}/>, div);
  });

  it('should match its snapshot with required props', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});