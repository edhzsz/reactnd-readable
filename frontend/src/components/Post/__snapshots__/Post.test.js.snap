// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Post should match its snapshot with required props 1`] = `
<div
  className="post"
>
  <div
    className="post-title base"
  >
    <span>
      Udacity is the best place to learn React
    </span>
  </div>
  <div
    className="post-contents"
  >
    <div
      className="post-votes"
    >
      <Votes
        size={30}
        votes={6}
      />
    </div>
    <div
      className="post-main"
    >
      <div
        className="post-body"
      >
        <p>
          Everyone says so after all.
        </p>
        <span
          className="post-author"
        >
          thingtwo
        </span>
        <span
          className="post-category"
        >
          react
        </span>
        <Timestamp
          className="post-timestamp"
          component="span"
          format="full"
          includeDay={false}
          precision={1}
          since={null}
          style={Object {}}
          time={1467166872.634}
          until={null}
          utc={true}
        />
      </div>
      <CommentsList
        comments={
          Array [
            Object {
              "author": "thingtwo",
              "body": "Hi there! I am a COMMENT.",
              "deleted": false,
              "id": "894tuq4ut84ut8v4t8wun89g",
              "parentDeleted": false,
              "parentId": "8xf0y6ziyjabvozdd253nd",
              "timestamp": 1468166872634,
              "voteScore": 6,
            },
            Object {
              "author": "thingone",
              "body": "Comments. Are. Cool.",
              "deleted": false,
              "id": "8tu4bsun805n8un48ve89",
              "parentDeleted": false,
              "parentId": "8xf0y6ziyjabvozdd253nd",
              "timestamp": 1469479767190,
              "voteScore": -5,
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;
